src/angular/metadata.ts

```diff
@@ -44,3 +44,20 @@ export class ComponentMetadata extends DirectiveMetadata {
     super(controller, decorator, selector);
   }
 }
+
+export class PipeMetadata {
+  constructor(
+    public readonly controller: ts.ClassDeclaration,
+    public readonly decorator: ts.Decorator,
+    public readonly name?: string,
```

Comment:
```suggestion
    readonly name: string,
```

Reply:
Unfortunately, We should keep `?`.  `name` may be not present in the source